tools/resourcedocsgen/pkg/metadata.go

```diff
@@ -0,0 +1,44 @@
+package pkg
+
+// PackageStatus is a type to indicate a package's status.
+type PackageStatus string
+
+// PackageCategory is a type to indicate the category a package
+// belongs to.
+type PackageCategory string
+
+const (
+	// PackageStatusGA indicates that a package is generally available for use.
+	PackageStatusGA PackageStatus = "ga"
+	// PackageStatusPublicPreview indicates that a package is available as
+	// pre-release and can undergo changes before it goes GA.
+	PackageStatusPublicPreview PackageStatus = "public_preview"
+
+	PackageCategoryCloud    PackageCategory = "Cloud"
```

Comment:
This is not exhaustive at the moment and in fact all packages get categorized as Cloud instead of their respective categories as seen on our API Reference and Cloud Providers pages today.